homeassistant/components/switch/xiaomi_miio.py

```diff
@@ -154,20 +154,16 @@ def _try_command(self, mask_error, func, *args, **kwargs):
     @asyncio.coroutine
     def async_turn_on(self, **kwargs):
         """Turn the plug on."""
-        result = yield from self._try_command(
-            "Turning the plug on failed.", self._plug.on)
-
-        if result:
+        if (yield from self._try_command("Turning the plug on failed.",
+                                         self._plug.on)):
```

Comment:
Considering that this is not just saving the result to a single variable and copying it over afterwards, I prefer clean code over "cleverness" here, so I think this change (also below) should be reverted :-) 

If no one voices opinions after that I'll go ahead and merge this.